azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -282,6 +289,8 @@ public void run() {
         closeLogger();
         updateFlow();
       } finally {
+        reportFlowFinishedMetrics();
```

Comment:
Calling this in `finally` means that the flow status may not have been updated properly... is it wise to do it in `finally` *and* still rely on flow status to mark the right metric?

Reply:
I think is safe to place it here because in line 276 execution status is set to FAILED on the occurrence of any exceptions and any further failures won't change that.